modules/apps/jax-rs/mdm-admin/src/main/java/org/wso2/carbon/mdm/api/User.java

```diff
@@ -505,6 +505,39 @@ public Response getAllUsersByUsername(@QueryParam("username") String userName) t
     }
 
     /**
+     * Get the list of user names in the system.
+     *
+     * @return A list of user names.
+     * @throws MDMAPIException
+     */
+    @GET
+    @Path("users-by-username")
+    public Response getAllUserNamesByUsername(@QueryParam("username") String userName) throws MDMAPIException {
+        if (log.isDebugEnabled()) {
+            log.debug("Getting the list of users by name");
+        }
+        UserStoreManager userStoreManager = MDMAPIUtils.getUserStoreManager();
+        ArrayList<String> userList = new ArrayList<String>();
```

Comment:
As a best practice, use appropriate abstractions for variable definitions for better extensibility code the code.